server/models/postgis/user.py

```diff
@@ -0,0 +1,66 @@
+#import geojson
+from server import db
+#from server.models.dtos.user_dto import UserDTO, UserFilterDTO, Pagination, UserSearchQuery, UserSearchDTO, \
+#     ListedUser
+from server.models.dtos.user_dto import UserDTO
+from server.models.postgis.lookups import UserRole
+from server.models.postgis.utils import NotFound, timestamp
+
+
+class User(db.Model):
+    """ Describes the user account table """
+    __tablename__ = "dmis_users"
+
+    user_id = db.Column(db.BigInteger, primary_key=True, index=True)
+    username = db.Column(db.String, unique=True, index=True)
+    role = db.Column(db.Integer, default=0, nullable=False)
+    email_address = db.Column(db.String)
+    password = db.Column(db.String)
+
+    def create(self):
+        """ Creates and saves the current model to the DB """
+        db.session.add(self)
+        db.session.commit()
+
+    def save(self):
+        db.session.commit()
+
+    def get_by_id(self, user_id: int):
+        """ Return the user for the specified id, or None if not found """
+        return User.query.get(user_id)
+
+    def get_by_username(self, username: str):
+        """ Return the user for the specified username, or None if not found """
+        return User.query.filter_by(username=username).one_or_none()
+
+    def update(self, user_dto: UserDTO):
+        """ Update the user details """
+        self.email_address = user_dto.email_address.lower() if user_dto.email_address else None
+        db.session.commit()
+
+    def set_password(self, hashed_password: str):
```

Comment:
You shouldn't need separate setters for password and role, they can be set on the model and then saved in a oner

So set_password, set_role can be deleted